packages/libs/core/src/route/rewrite.ts

```diff
@@ -27,6 +30,21 @@ export function getRewritePath(
       return;
     }
 
+    // No-op rewrite support for pages: skip to next rewrite if path does not map to existing non-dynamic and dynamic routes
+    if (pageManifest && path === destination) {
+      const url = handlePageReq(
```

Comment:
What about public files?

Reply:
I suppose those just work because they get checked earlier?

Reply:
Sure, we can add check for static files too.

Reply:
Ah yes, I believe public files cannot be rewritten unless you use the new objects format:

> Rewrites are applied after checking the filesystem (pages and /public files) and before dynamic routes by default. This behavior can be changed by instead returning an object instead of an array from the rewrites function since v10.1 of Next.js:

Reply:
Also now that the code chunking logic is much better from Next.js, we could also start thinking about combining default/API lambdas to simplify rewrites and also support cross-rewrites (between default an API lambdas)

Reply:
Yeah, I was about to mention api routes. But this is surprisingly simple 👍🏽